### PR TITLE
Make BiomeModifications datapack-compatible.

### PR DESCRIPTION
**Pull request by peaceheis**
All this does is add a check to the vanilla dimension BiomeSelectors to check tags. Backwards compat is guaranteed, as an or has been used (I believe the Nether selector in particular needs the original check).

### CODE REVIEW

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 1]
```diff
@@ -69,25 +72,25 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether,
-	 * assuming Vanilla's default multi noise biome source with the nether preset is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether. Datapack Compatible.
 	 * <p>
 	 * This selector will also match modded biomes that have been added to the nether using {@link NetherBiomes}.
 	 */
 	public static Predicate<BiomeSelectionContext> foundInTheNether() {
-		return context -> context.canGenerateIn(DimensionOptions.NETHER);
+		TagKey<Biome> NETHER = TagKey.of(Registry.BIOME_KEY, new Identifier("minecraft", "is_nether"));
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 2]
```diff
@@ -69,25 +72,25 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the End,
-	 * assuming Vanilla's default End biome source is used.
+	 * Returns a biome selector that will match all biomes that spawn in the end. Datapack compatible.
 	 */
 	public static Predicate<BiomeSelectionContext> foundInTheEnd() {
-		return context -> context.canGenerateIn(DimensionOptions.END);
+		TagKey<Biome> END = TagKey.of(Registry.BIOME_KEY, new Identifier("minecraft", "is_end"));
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
```

**Review by LambdAurora — COMMENTED** [review]
I think you have some left-overs.

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 3]
```diff
@@ -69,25 +70,23 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
 	 */
 	public static Predicate<BiomeSelectionContext> foundInOverworld() {
-		return context -> context.canGenerateIn(DimensionOptions.OVERWORLD);
+		return context -> context.isIn(BiomeTags.IS_OVERWORLD) || context.canGenerateIn(DimensionOptions.OVERWORLD);
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether,
-	 * assuming Vanilla's default multi noise biome source with the nether preset is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether. Datapack Compatible.
```
Comment (EnnuiL) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether.
	 * This method is compatible with biomes defined on the {@code minecraft:is_nether} tag.
```
If the "Assuming Vanilla's source" line is still true, then it should be readded

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 4]
```diff
@@ -69,25 +70,23 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
 	 */
 	public static Predicate<BiomeSelectionContext> foundInTheNether() {
-		return context -> context.canGenerateIn(DimensionOptions.NETHER);
+		return context -> context.isIn(BiomeTags.IS_NETHER) || context.canGenerateIn(DimensionOptions.NETHER);
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the End,
-	 * assuming Vanilla's default End biome source is used.
+	 * Returns a biome selector that will match all biomes that spawn in the end. Datapack compatible.
```
Comment (EnnuiL) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
	 * Returns a biome selector that will match all biomes that would normally spawn in the End.
	 * This method is compatible with biomes defined on the {@code minecraft:is_end} tag.
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 5]
```diff
@@ -69,25 +70,23 @@ public static Predicate<BiomeSelectionContext> vanilla() {
 	 * assuming Vanilla's default biome source is used.
```
Comment (EnnuiL) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]: The Javadoc for foundInOverworld hasn't been updated; It should have the same "This method is compatible with this tag" comment as the other two

**Review by EnnuiL — CHANGES_REQUESTED** [review]

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeModification.java [hunk 6]
```diff
@@ -51,8 +51,8 @@ public BiomeModification add(ModificationPhase phase, Predicate<BiomeSelectionCo
 	/**
 	 * Adds a modifier that is sensitive to the current state of the biome when it is applied.
 	 * Examples for this are modifiers that apply scales to existing values (e.g. half the temperature).
-	 * <p>
-	 * For modifiers that should only be applied if a given condition is met for a Biome, please add these
+	 *
+	 * <p>For modifiers that should only be applied if a given condition is met for a Biome, please add these
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeModification.java]:
```suggestion
	 * <p>
	 * For modifiers that should only be applied if a given condition is met for a Biome, please add these
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java [hunk 7]
```diff
@@ -33,30 +33,30 @@ public enum ModificationPhase {
...
+	 * <p><b>Examples:</b> Replace mineshafts with biome-specific mineshafts
 	 */
 	REPLACEMENTS,
 
 	/**
 	 * The appropriate phase for modifiers that perform wide-reaching biome postprocessing.
-	 * <p>
-	 * <b>Examples:</b> Mods that allow modpack authors to customize world generation, changing biome
+	 *
+	 * <p><b>Examples:</b> Mods that allow modpack authors to customize world generation, changing biome
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java]:
```suggestion
	 * <p>
	 * <b>Examples:</b> Mods that allow modpack authors to customize world generation, changing biome
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java [hunk 8]
```diff
@@ -33,30 +33,30 @@ public enum ModificationPhase {
...
+	 * <p><b>Examples:</b> Remove iron ore from plains, remove ghasts
 	 */
 	REMOVALS,
 
 	/**
 	 * The appropriate phase for modifiers that replace existing features with modified features.
-	 * <p>
-	 * <b>Examples:</b> Replace mineshafts with biome-specific mineshafts
+	 *
+	 * <p><b>Examples:</b> Replace mineshafts with biome-specific mineshafts
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java]:
```suggestion
	 * <p>
	 * <b>Examples:</b> Replace mineshafts with biome-specific mineshafts
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java [hunk 9]
```diff
@@ -33,30 +33,30 @@ public enum ModificationPhase {
...
 	 */
 	ADDITIONS,
 
 	/**
 	 * The appropriate phase for modifiers that remove features or other aspects of biomes (i.e. removal of spawns,
 	 * removal of features, etc.).
-	 * <p>
-	 * <b>Examples:</b> Remove iron ore from plains, remove ghasts
+	 *
+	 * <p><b>Examples:</b> Remove iron ore from plains, remove ghasts
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java]:
```suggestion
	 * <p>
	 * <b>Examples:</b> Remove iron ore from plains, remove ghasts
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java [hunk 10]
```diff
@@ -33,30 +33,30 @@ public enum ModificationPhase {
 	/**
 	 * The appropriate phase for enriching biomes by adding to them without relying on
 	 * other information in the biome, or removing other features.
-	 * <p>
-	 * <b>Examples:</b> New ores, new vegetation, new structures
+	 *
+	 * <p><b>Examples:</b> New ores, new vegetation, new structures
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/ModificationPhase.java]:
```suggestion
	 * <p>
	 * <b>Examples:</b> New ores, new vegetation, new structures
```

**Review by LambdAurora — COMMENTED** [review]
I am gonna make checkstyle explode.

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 11]
```diff
@@ -65,29 +66,29 @@ public static Predicate<BiomeSelectionContext> vanilla() {
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the Overworld,
-	 * assuming Vanilla's default biome source is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the Overworld.
+	 * This method is compatible with biomes defined on the {@code minecraft:is_overworld} tag.
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
	 * This method is compatible with biomes defined in the {@code minecraft:is_overworld} tag.
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 12]
```diff
@@ -65,29 +66,29 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
 	public static Predicate<BiomeSelectionContext> foundInOverworld() {
-		return context -> context.canGenerateIn(DimensionOptions.OVERWORLD);
+		return context -> context.isIn(BiomeTags.IS_OVERWORLD) || context.canGenerateIn(DimensionOptions.OVERWORLD);
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether,
-	 * assuming Vanilla's default multi noise biome source with the nether preset is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the Nether.
+	 * This method is compatible with biomes defined on the {@code minecraft:is_nether} tag.
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
	 * This method is compatible with biomes defined in the {@code minecraft:is_nether} tag.
```

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 13]
```diff
@@ -65,29 +66,29 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
 	public static Predicate<BiomeSelectionContext> foundInTheNether() {
-		return context -> context.canGenerateIn(DimensionOptions.NETHER);
+		return context -> context.isIn(BiomeTags.IS_NETHER) || context.canGenerateIn(DimensionOptions.NETHER);
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the End,
-	 * assuming Vanilla's default End biome source is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the End.
+	 * This method is compatible with biomes defined on the {@code minecraft:is_end} tag.
```
Comment (LambdAurora) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]:
```suggestion
	 * This method is compatible with biomes defined in the {@code minecraft:is_end} tag.
```

**Review by LambdAurora — APPROVED** [review]

**Review by Leo40Git — APPROVED** [review]

**Review by oliviathevampire — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]
Fix the other suggestions and it'll be good

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 14]
```diff
@@ -65,29 +66,30 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
-		return context -> context.canGenerateIn(DimensionOptions.NETHER);
+		return context -> context.isIn(BiomeTags.IS_NETHER) || context.canGenerateIn(DimensionOptions.NETHER);
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the End,
-	 * assuming Vanilla's default End biome source is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the End.
+	 * This method is compatible with biomes defined in the {@code minecraft:is_end} tag.
+	 * his selector will also match modded biomes that have been added to the End using {@link TheEndBiomes}.
```
Comment (EnnuiL) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]: <p> for consistency with `foundInTheNether`, and also T

```suggestion
         * <p>
	 * This selector will also match modded biomes that have been added to the End using {@link TheEndBiomes}.
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]

File: library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java [hunk 15]
```diff
@@ -65,29 +66,31 @@ public static Predicate<BiomeSelectionContext> vanilla() {
...
-		return context -> context.canGenerateIn(DimensionOptions.NETHER);
+		return context -> context.isIn(BiomeTags.IS_NETHER) || context.canGenerateIn(DimensionOptions.NETHER);
 	}
 
 	/**
-	 * Returns a biome selector that will match all biomes that would normally spawn in the End,
-	 * assuming Vanilla's default End biome source is used.
+	 * Returns a biome selector that will match all biomes that would normally spawn in the End.
+	 * This method is compatible with biomes defined in the {@code minecraft:is_end} tag.
+     * <p>
```
Comment (EnnuiL) [inline library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/BiomeSelectors.java]: i screwed up at the suggestion thing
```suggestion
   * <p>
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]

**Review by EnnuiL — APPROVED** [review]
Stamped with the Ennui's ~~Making Lambda Stop Worrying About Further Delays 'Til Approval~~ Stamp of Approval

**Pull request merged**